src/promptflow/promptflow/executor/_flow_nodes_scheduler.py

```diff
@@ -21,34 +21,40 @@
 
 
 class FlowNodesScheduler:
-    def __init__(self, tools_manager: ToolsManager) -> None:
+    def __init__(
+        self,
+        tools_manager: ToolsManager,
+        inputs: Dict,
+        nodes_from_invoker: List[Node],
+        node_concurrency: int,
+        context: FlowExecutionContext,
+    ) -> None:
         self.tools_manager = tools_manager
+        self.future_to_node: Dict[Future, Node] = {}
+        self._node_concurrency = min(node_concurrency, DEFAULT_CONCURRENCY_FLOW)
+        flow_logger.info(f"Start to run {len(nodes_from_invoker)} nodes with concurrency level {node_concurrency}.")
+        self.dag_manager = DAGManager(nodes_from_invoker, inputs)
+        self.context = context
 
     def execute(
-        self, context: FlowExecutionContext, inputs: Dict, nodes_from_invoker: List[Node], node_concurrency: int
+        self,
     ) -> Tuple[dict, dict]:
-        dag_manager = DAGManager(nodes_from_invoker, inputs)
-        node_concurrency = min(node_concurrency, DEFAULT_CONCURRENCY_FLOW)
-        logger.info(f"Start to run {len(nodes_from_invoker)} nodes with concurrency level {node_concurrency}.")
 
-        self.context = context
-        self.inputs = inputs
         parent_context = contextvars.copy_context()
         with ThreadPoolExecutor(
-            max_workers=node_concurrency, initializer=set_context, initargs=(parent_context,)
+            max_workers=self._node_concurrency, initializer=set_context, initargs=(parent_context,)
         ) as executor:
-            self.future_to_node: Dict[Future, Node] = {}
-            self._execute_nodes(dag_manager, executor)
+            self._execute_nodes(executor)
 
-            while not dag_manager.completed():
+            while not self.dag_manager.completed():
                 try:
                     if not self.future_to_node:
                         raise NoNodeExecutedError("No nodes are ready for execution, but the flow is not completed.")
                     completed_futures, _ = futures.wait(self.future_to_node.keys(), return_when=futures.FIRST_COMPLETED)
-                    dag_manager.complete_nodes(self._collect_outputs(completed_futures))
+                    self.dag_manager.complete_nodes(self._collect_outputs(completed_futures))
                     for each_future in completed_futures:
                         del self.future_to_node[each_future]
-                    self._execute_nodes(dag_manager, executor)
+                    self._execute_nodes(executor)
                 except Exception as e:
                     for unfinished_future in self.future_to_node.keys():
                         node_name = self.future_to_node[unfinished_future].name
```

Comment:
Can we print a log outside the for loop to reduce logs? Just print the list of node names.